### ISSUE TITLE
Loading background-image from CSS

### ISSUE THREAD

**Issue by FlorianRappl** [AngleSharp/AngleSharp.Css#4]
See [AngleSharp #188](https://github.com/AngleSharp/AngleSharp/issues/188) for full discussion.

Such as recursive downloads (or similar CSSOM oriented downloads), the resources requested in values should also be handed over to the corresponding requesters - if there are any, and only if resource loading is activate!

**Comment by GieltjE** [comment]
We used to do (roughly) something like the following:

```            ICssStyleSheet result = cssParser.ParseStyleSheet(toReturn);
            foreach (ICssRule cssRule in result.Rules)
            {
                switch (cssRule.Type)
                {
                    case CssRuleType.Import:
                        break;
                    case CssRuleType.Style:
                        foreach (ICssNode cssNode in ((ICssStyleRule)cssRule).Children)
                        ((ICssStyleDeclaration)cssNode).BackgroundImage```

We extract some images and fonts url's, which is the only point to our usage of this library, is there currently a way to achieve this?

**Comment by FlorianRappl** [comment]
The `ICssStyleDeclaration`  has extension methods such as `SetBackgroundImage` (or `GetBackgroundImage`). These methods are the replacement for the earlier (hard-coded) properties. We would love to use properties, but we need an extension capability, hence as long as there are no "extension properties" we need to have the methods explicitly.

The value model is also there to help you (but currently not exposed for a reason - it is not fully finished and has too many rough edges) and we will bring in automatic loading of these resources as well soon.

Just hang in tight (and ask if anything is unclear)!

TL;DR: It should be possible to achieve all you did beforehand (and more) with (AngleSharp.Css) v0.10 as well, but some things may have changed slightly.

HTH!

**Comment by FlorianRappl** [comment]
Landed in `devel` and in the preview.

Works in conjunction with the render tree, which can be extracted using using `Render` on an `IWindow`.

Full example:

```cs
var tree = document.DefaultView.Render();
var node = tree.Find(document.QuerySelector("div"));
await node.DownloadResources();
```
